src/server/pfs/server/url.go

```diff
@@ -117,17 +132,24 @@ func putFileURLRecursive(ctx context.Context, taskService task.Service, uw *file
 			}
 			return nil
 		}
-		if err := objClient.Walk(ctx, url.Object, func(p string) error {
+		list := bucket.List(&blob.ListOptions{Prefix: url.Object})
+		var listObj *blob.ListObject
+		for {
+			listObj, err = list.Next(ctx)
+			if err != nil {
+				if err == io.EOF {
```

Comment:
We might want this to be an `errors.Is` check.

Reply:
The example in the [godocs ](https://pkg.go.dev/gocloud.dev/blob@v0.27.0#example-Bucket.List) for `.List()` uses io.EOF so I think we should leave it this way

Reply:
That isn't the recommended way to perform error checks anymore (https://pkg.go.dev/errors). We've had errors in the past using equality checks, so it would be better to future proof against that.